### DISCUSSION TITLE
database tables

### DISCUSSION THREAD

**Discussion by SinisterSpatula** [DSJAS/DSJAS#16] [Ideas]
It would be great if we could have a "VanityAccountNumber" that can be used for themes.  I know internally it needs to be tracked by "Account: 000001" but that account could also have a vanity number like "****1382".  This way the vanity number could be anything and not mess with the transferring system and still look more real.  Also, for transaction types, can we have "deposit" type added to the existing types?  And finally, could we also track the "Available Balance" per each separate transaction, so there is a history of balance over time that can be used by themes.

Also for the users/accounts database table:  if it could be expanded to contain the users phone number, that way my fake two-factor auth could use that to appear more real.  Otherwise I just have to generate a random phone number to display which could look less real.  Storing their Address street, city, state, etc could also be used for display by themes.

**Comment by ejv2** [comment]
Fake account numbers are something I have had on the heap of things to do for a little while, as it is kind of noticeable that you appear to own three bank accounts, all of which have single digit numbers and are consecutive (not particularly likely to actually happen). This wouldn't be particularly hard to implement, as it just involves an update to the databases and their columns, but I may also need to do some shifting around with how themes handle things.

As for more information stored as part of the user account records, that is totally fine too. But, if you want to retrieve it using a module, you may need to wait until I have finished planning how the module API will work. Eventually, there will be a function to get information for the current user, but I haven't planned even the basics yet. You may have to sit tight for this one, unfortunately.

**Comment by ejv2** [comment]
Ok. I've drafted an initial version of the "Vanity account number" system and will push it later tonight, once I've double checked everything for regressions it caused.

So far, this is what it looks like:

![Screenshot of the new system](https://i.imgur.com/q8Xlq2J.png "DSJAS Dashboard Screenshot")

At the moment, there is no way to modify this value from the user's point of view: it is randomly generated and randomly entered by the DSJAS core. Would you be interested in the user being able to modify the value, or is that a bit redundant, as long as the value makes sense?

Also, currently, the value can range from "111111111" to "999999999", meaning it is impossible for a value like "1" or "10" to appear. Do you think that will be realistic enough?

**Reply by ejv2** [reply]
Current finalised version introduced in [10f32f3](https://github.com/DSJAS/DSJAS/commit/10f32f3a5e79f6a420b56d71082f17797b560e42) and finalised in [4aee89e](https://github.com/DSJAS/DSJAS/commit/4aee89e83de9844f64236adcc35e9fedd7a4b268).

I'm happy with this. Feedback welcome and appreciated. This will be shipped in the next update, which will (hopefully) also include statistics.

**Reply by SinisterSpatula** [reply]
Looks awesome!  Maybe in the future the admin panel can allow changing the vanity number, if not, they could also just use something like phpmyadmin to modify directly in the database.
